api/src/components/reference/__tests__/createReference.test.ts

```diff
@@ -0,0 +1,122 @@
+import * as testUtils from 'lib/testUtils';
+
+describe('create a reference', () => {
+    beforeEach(async () => {
+        await testUtils.clearDB();
+        await testUtils.testSeed();
+    });
+
+    test('User can add a reference to their DRAFT publication', async () => {
+        const reference = await testUtils.agent
+            .post('/publications/publication-problem-draft/reference')
+            .query({ apiKey: '000000005' })
+            .send({
+                id: '0001',
+                location: 'http://octopus.ac',
+                publicationId: 'publication-problem-draft',
+                text: '<p>Reference 1</p>',
+                type: 'TEXT'
+            });
+
+        expect(reference.status).toEqual(200);
+    });
+
+    test('User cannot add a reference to their LIVE publication', async () => {
+        const reference = await testUtils.agent
+            .post('/publications/publication-problem-live/reference')
+            .query({ apiKey: '123456789' })
+            .send({
+                id: '0001',
+                location: null,
+                publicationId: 'publication-problem-live',
+                text: '<p>Reference 1</p>',
+                type: 'TEXT'
+            });
+
+        expect(reference.status).toEqual(403);
+    });
+    test('User cannot add a reference to another users DRAFT publication', async () => {
+        const reference = await testUtils.agent
+            .post('/publications/publication-problem-draft/reference')
+            .query({ apiKey: '987654321' })
```

Comment:
a comment on the existing tests that these test don't address (we might have to add to backlog I think) - the apiKey is a work around really (and only used by 1 user - octopus), it does not really represent a true 'user' (as that uses a decoded Bearer token), we can argue their equivalence but all the API tests for users will give me less confidence in what they are testing and also what they might be missing...